src/controllers/restatecluster/reconcilers/compute.rs

```diff
@@ -229,21 +229,49 @@ fn env(cluster_name: &str, custom: Option<&[EnvVar]>) -> Vec<EnvVar> {
             value_from: None,
         });
 
-    let defaults = Some(EnvVar {
-        name: "POD_NAME".into(),
-        value: None,
-        value_from: Some(EnvVarSource {
-            config_map_key_ref: None,
-            field_ref: Some(ObjectFieldSelector {
-                api_version: None,
-                field_path: "metadata.name".into(),
+    let downward_api_vars = [
+        EnvVar {
+            name: "POD_NAME".into(),
+            value: None,
+            value_from: Some(EnvVarSource {
+                config_map_key_ref: None,
+                field_ref: Some(ObjectFieldSelector {
+                    api_version: None,
+                    field_path: "metadata.name".into(),
+                }),
+                resource_field_ref: None,
+                secret_key_ref: None,
             }),
-            resource_field_ref: None,
-            secret_key_ref: None,
-        }),
-    })
-    .into_iter()
-    .chain(defaults);
+        },
+        EnvVar {
+            name: "POD_ZONE".into(),
+            value: None,
+            value_from: Some(EnvVarSource {
+                config_map_key_ref: None,
+                field_ref: Some(ObjectFieldSelector {
+                    api_version: None,
+                    field_path: "metadata.labels['topology.kubernetes.io/zone']".into(),
```

Comment:
fyi, this pod label doesn't currently exist for pods, its a host label. it will be available in near-future GA k8s versions though though https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/4742-node-topology-downward-api/README.md